tm/src/main/java/io/seata/tm/api/transaction/TransactionInfo.java

```diff
@@ -60,8 +60,16 @@ public boolean rollbackOn(Throwable ex) {
 
         RollbackRule winner = null;
         int deepest = Integer.MAX_VALUE;
+        boolean setRollback = false;
 
         if (this.rollbackRules != null) {
+            for (RollbackRule rule : this.rollbackRules){
+                if(rule instanceof NoRollbackRule){
+                    continue;
+                }
+                setRollback = true;
```

Comment:
execute many times?

Reply:
done

Reply:
What you have modified and what you originally expressed, I can’t see any changes, the code is more complicated.